sdk/src/main/scala/com/horizen/certificatesubmitter/keys/SchnorrKeysSignaturesListBytes.scala

```diff
@@ -0,0 +1,44 @@
+package com.horizen.certificatesubmitter.keys
+
+import com.horizen.schnorrnative.{SchnorrPublicKey, SchnorrSignature, ValidatorKeysUpdatesList}
+
+import java.util
+
+case class SchnorrKeysSignaturesListBytes(
+                                      schnorrSignersPublicKeysBytesList: Seq[Array[Byte]],
+                                      schnorrMastersPublicKeysBytesList: Seq[Array[Byte]],
+                                      newSchnorrSignersPublicKeysBytesList: Seq[Array[Byte]],
+                                      newSchnorrMastersPublicKeysBytesList: Seq[Array[Byte]],
+                                      updatedSigningKeysSkSignatures: Seq[Option[Array[Byte]]],
+                                      updatedSigningKeysMkSignatures: Seq[Option[Array[Byte]]],
+                                      updatedMasterKeysSkSignatures: Seq[Option[Array[Byte]]],
+                                      updatedMasterKeysMkSignatures: Seq[Option[Array[Byte]]],
+                                    )
+
+object SchnorrKeysSignaturesListBytes{
+  def getSchnorrKeysSignaturesList(schnorrKeysSignaturesListBytes: SchnorrKeysSignaturesListBytes): ValidatorKeysUpdatesList = {
+    new ValidatorKeysUpdatesList(
+      byteArrayToKeysList(schnorrKeysSignaturesListBytes.schnorrSignersPublicKeysBytesList),
+      byteArrayToKeysList(schnorrKeysSignaturesListBytes.schnorrMastersPublicKeysBytesList),
+      byteArrayToKeysList(schnorrKeysSignaturesListBytes.newSchnorrSignersPublicKeysBytesList),
+      byteArrayToKeysList(schnorrKeysSignaturesListBytes.newSchnorrMastersPublicKeysBytesList),
+      byteArrayToSignaturesList(schnorrKeysSignaturesListBytes.updatedSigningKeysSkSignatures),
+      byteArrayToSignaturesList(schnorrKeysSignaturesListBytes.updatedSigningKeysMkSignatures),
+      byteArrayToSignaturesList(schnorrKeysSignaturesListBytes.updatedMasterKeysSkSignatures),
+      byteArrayToSignaturesList(schnorrKeysSignaturesListBytes.updatedMasterKeysMkSignatures),
+      schnorrKeysSignaturesListBytes.schnorrSignersPublicKeysBytesList.size
+    )
+  }
+
+  private def byteArrayToKeysList(schnorrPublicKeysBytesList: Seq[Array[Byte]]): util.List[SchnorrPublicKey] =
+    scala.collection.JavaConverters.seqAsJavaList(schnorrPublicKeysBytesList.map(SchnorrPublicKey.deserialize))
```

Comment:
note: here we create a native object, be sure that in all the places where we use it the memory was deallocated after. I saw at least one place in `ThresholdSignatureCircuitWithKeyRotationImplZendoo.createProof` where we didn't "close" it. Check other places.

PS: this method also must be moved to `ThresholdSignatureCircuitWithKeyRotationImplZendoo` class

Reply:
Done